03_Modules/EVDriver/src/module/LocalAuthListService.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright Contributors to the CitrineOS Project
+//
+// SPDX-License-Identifier: Apache 2.0
+
+import { AttributeEnumType, SendLocalListRequest, UpdateEnumType } from "@citrineos/base";
+import { IDeviceModelRepository, ILocalAuthListRepository, VariableCharacteristics, VariableAttribute, LocalListVersion, SendLocalList, } from "@citrineos/data";
+import { LocalListAuthorization } from "@citrineos/data/src/layers/sequelize/model/Authorization/LocalListAuthorization";
+
+import { v4 as uuidv4 } from 'uuid';
+
+export class LocalAuthListService {
+  protected _localAuthListRepository: ILocalAuthListRepository;
+  protected _deviceModelRepository: IDeviceModelRepository;
+
+  constructor(localAuthListRepository: ILocalAuthListRepository,
+    deviceModelRepository: IDeviceModelRepository
+  ) {
+    this._localAuthListRepository = localAuthListRepository;
+    this._deviceModelRepository = deviceModelRepository;
+  }
+
+  /**
+   * Validates a SendLocalListRequest and persists it, then returns the correlation Id.
+   *
+   * @param {string} stationId - The ID of the station to which the SendLocalListRequest belongs.
+   * @param {string} correlationId - The correlation Id that will be used for the SendLocalListRequest.
+   * @param {SendLocalListRequest} sendLocalListRequest - The SendLocalListRequest to validate and persist.
+   * @return {SendLocalList} The persisted SendLocalList.
+   */
+  async persistSendLocalListForStationIdAndCorrelationIdAndSendLocalListRequest(stationId: string, correlationId: string, sendLocalListRequest: SendLocalListRequest): Promise<SendLocalList> {
+    const localListVersion = await this._localAuthListRepository.readOnlyOneByQuery({
+      where: {
+        stationId: stationId,
+      },
+      include: [LocalListAuthorization],
+    });
+    const sendLocalList = await this.createSendLocalListFromStationIdAndRequestAndCurrentVersion(stationId, correlationId, sendLocalListRequest, localListVersion);
+
+    const newLocalAuthListLength = await this.countUpdatedAuthListFromRequestAndCurrentVersion(sendLocalList, localListVersion);
+    // TODO If Device Model is updated to allow different variable characteristics for the same variable per station, then we need to update this
```

Comment:
Can we change this away from a TODO?

Reply:
What did you have in mind? This being a TODO will make it easier to spot cases that need to be updated when we finally overhaul the DeviceModel.

Reply:
My understanding was to get away from TODOs and find a different method of pointing out things that will be improved. It's also a bit misleading right now. If I want to resolve the TODO I can't really yet. If we make it a different comment that we can make searchable that would already be fine by me. E.g. `DEVICE MODEL UPDATE: required action...`